### PR TITLE
NEST NEural Simulation Tool

### PR DESCRIPTION
**Pull request by physicalist**
NEST is a simulator for spiking neural network models that focuses on the dynamics, size and structure of neural systems rather than on the exact morphology of individual neurons.

See http://nest-initiative.org/

### CODE REVIEW

File: nest.rb [hunk 1]
```diff
@@ -0,0 +1,86 @@
+require 'formula'
+
+# NEST is the NEural Simulation Tool
+
```
Comment (iml) [inline nest.rb]: The above lines are unnecessary. Also please use all double-quotes with Ruby code.

File: nest.rb [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+class Nest < Formula
+  homepage 'http://www.nest-initiative.org'
+  url 'http://www.nest-initiative.org/download/gplreleases/nest-2.4.2.tar.gz'
+  sha1 '86da83c9747c898e63d00a2f99159700a68d8b36'
+
+  option "without-python", "Do not build python bindings (saves some time but not recommended; pyNN wouldn't work, among other things)"
+  option "without-gsl", "Don't use Gnu scientific library (GSL) for extended numerical possibilites (NOT recommended, WILL break some neuron models)"
+
+  option "enable-pthread", "Use pthreads"
+  option "enable-openmpi", "Use OpenMPI"
```
Comment (iml) [inline nest.rb]: Options should be `with-pthread` or `without-pthread` such that the DSL is `build.with?` or `build.without?` later on. `build.include?` is deprecated.

File: nest.rb [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+  end
+
+  test do
+    # simple check whether NEST was compiled & linked
+    system "#{bin}/nest", "--version"
+
+    # use pyNEST's built-in test suite
+    python do
+      system python, "-c", "import nest; nest.test()"
+    end
```
Comment (iml) [inline nest.rb]:
```ruby
Language::Python.each_python(build) do |python, _version|
  system python, "-c", "import nest; nest.test()"
end

File: nest.rb (hunk 1 above)
Comment (physicalist) [inline nest.rb]: When I created the formula, `require` was still required :)

**Comment by dpo** [comment]
@physicalist Will this be updated?

File: nest.rb [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+
+# NEST is the NEural Simulation Tool
+
+class Nest < Formula
+  homepage 'http://www.nest-initiative.org'
+  url 'http://www.nest-initiative.org/download/gplreleases/nest-2.4.2.tar.gz'
+  sha1 '86da83c9747c898e63d00a2f99159700a68d8b36'
+
+  option "without-python", "Do not build python bindings (saves some time but not recommended; pyNN wouldn't work, among other things)"
+  option "without-gsl", "Don't use Gnu scientific library (GSL) for extended numerical possibilites (NOT recommended, WILL break some neuron models)"
```
Comment (dpo) [inline nest.rb]: These options are redundant with the `depends_on` lines below.

File: nest.rb (hunk 2 above)
Comment (dpo) [inline nest.rb]: Please remove the `openmpi` option.

File: nest.rb [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+      > error: field of type 'std::vector<double>::const_iterator'
+      > (aka '__wrap_iter<const_pointer>') has private constructor
+      > next_(0),
+      > ^
+     EOS
+  end
+
+  depends_on :python => :recommended
+  depends_on "gsl" => :recommended
+  depends_on "open-mpi" if build.include? 'enable-openmpi'
```
Comment (dpo) [inline nest.rb]: Shoud be `depends_on :mpi`.

File: nest.rb [hunk 6]
```diff
@@ -0,0 +1,86 @@
...
+
+  depends_on "homebrew/python/scipy" if build.include? "python"
```
Comment (dpo) [inline nest.rb]: `build.with? "python"`

**Pull request closed without merging**